File: src/main/java/com/company/erp/features/auth/entity/RefreshToken.java
package com.company.erp.features.auth.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.Instant;

/**
 * Server-side record of an issued refresh token, identified by its {@code jti}
 * claim. {@code revokedAt} non-null invalidates the token; rotation on
 * {@code /auth/refresh} revokes the old row and creates a new one.
 */
@Getter
@Setter
@NoArgsConstructor
@Entity
@Table(name = "refresh_tokens")
public class RefreshToken {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(nullable = false, updatable = false)
    private Long id;

    @Column(name = "user_id", nullable = false)
    private Long userId;

    @Column(nullable = false, unique = true, length = 64)
    private String jti;

    @Column(name = "issued_at", nullable = false)
    private Instant issuedAt;

    @Column(name = "expires_at", nullable = false)
    private Instant expiresAt;

    @Column(name = "revoked_at")
    private Instant revokedAt;

    public boolean isActive(Instant now) {
        return revokedAt == null && expiresAt.isAfter(now);
    }
}

File: app/Features/Auth/Models/RefreshToken.php
<?php

namespace App\Features\Auth\Models;

use Carbon\CarbonInterface;
use Illuminate\Database\Eloquent\Model;

/**
 * Server-side record of an issued refresh token, identified by its jti claim.
 * A non-null revoked_at invalidates the token; rotation on /auth/refresh
 * revokes the old row and creates a new one. Port of the Spring RefreshToken
 * entity (no created/updated audit columns — its own issued_at/expires_at
 * carry the lifecycle).
 *
 * @property int $id
 * @property int $user_id
 * @property string $jti
 * @property \Illuminate\Support\Carbon $issued_at
 * @property \Illuminate\Support\Carbon $expires_at
 * @property \Illuminate\Support\Carbon|null $revoked_at
 */
class RefreshToken extends Model
{
    public $timestamps = false;

    protected $table = 'refresh_tokens';

    protected $fillable = ['user_id', 'jti', 'issued_at', 'expires_at', 'revoked_at'];

    protected $casts = [
        'issued_at' => 'datetime',
        'expires_at' => 'datetime',
        'revoked_at' => 'datetime',
    ];

    public function isActive(CarbonInterface $now): bool
    {
        return $this->revoked_at === null && $this->expires_at->greaterThan($now);
    }
}
